### PR TITLE
#67 test clean

### PR DESCRIPTION
**Pull request by SurinaCS**
Testing for clean.py. Covering the lambda handler deleting objects and covering edge cases such as no files found or files being exactly on retention day limit. Coverage is 100%!!

Addressing issue #67

### CODE REVIEW

File: clean/test_clean.py [hunk 1]
```diff
@@ -0,0 +1,195 @@
+"""Test file for clean.py"""
+
+import pytest
+import os
```
Comment (RidwanHamid501) [inline clean/test_clean.py:4]: reorganise imports for better pylint score

File: clean/test_clean.py [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+from datetime import datetime, timezone, timedelta
+from unittest.mock import MagicMock, patch
+from botocore.exceptions import NoCredentialsError, PartialCredentialsError
+from clean import (s3_connection, lambda_handler)
+
+
+@patch.dict('os.environ', {'AWS_ACCESS_KEY_ID': 'fake_access_key', 'AWS_SECRET_ACCESS_KEY': 'fake_secret_key'})
```
Comment (RidwanHamid501) [inline clean/test_clean.py:11]: you can create a fixture to manage environment variables rather than repeating @patch.dict:

```
@pytest.fixture
def aws_env_vars():
    with patch.dict("os.environ", {"AWS_ACCESS_KEY_ID": "fake_access_key", "AWS_SECRET_ACCESS_KEY": "fake_secret_key"}):
        yield
```

**Review by RidwanHamid501 — APPROVED** [review]
Good work!
